### DISCUSSION TITLE
Redshift Connection via CLI

### DISCUSSION THREAD

**Discussion by ramansh03** [duckdb/duckdb#6009] [General]
I need to connect my DuckDB CLI to a Redshift database with some host, port, username, and password.
But I am not able to find my way out.
Please help!!

I really appreciate any help you can provide.

**Comment by Mytherin** [comment]
Have you tried using the [Postgres scanner](https://duckdb.org/2022/09/30/postgres-scanner.html)?

**Reply by ramansh03** [reply]
Not yet
Can you please brief me about this approach?

**Reply by Mause** [reply]
@ramansh03 please refer to the docs: https://duckdb.org/docs/extensions/postgres_scanner

**Reply by ramansh03** [reply]
@Mause Tried but failed!

**Reply by Mause** [reply]
@ramansh03 if you give us details, we'd be happy to help

**Reply by ramansh03** [reply]
@Mause 
I can't provide the exact details but I can give demo values for you to look into.
HOST: abc123.ci3gimsawmqt.us-east-2.redshift.amazonaws.com 
PORT: 5439
DBNAME: rptdb
SCHEMA: biusr
USER: awsuser
PASSWORD: password123

**Reply by Mause** [reply]
I meant details on the failure? Ideally an error message, if possible

**Reply by ramansh03** [reply]
@Mause 
Okay, I can provide
I tried running this command :
**CALL postgres_attach('dbname=rptdb_ingest user=awsuser host=redshift-ingest-1.cfzy7kzssrme.ap-south-1.redshift.amazonaws.com:5439', source_schema='biusr');**

Error:
**Error: IO Error: Unable to connect to Postgres at dbname=rptdb_ingest user=awsuser host=redshift-ingest-1.cfzy7kzssrme.ap-south-1.redshift.amazonaws.com:5439: could not translate host name "redshift-ingest-1.cfzy7kzssrme.ap-south-1.redshift.amazonaws.com:5439" to address: Name or service not known**

**Reply by Mause** [reply]
Your redshift instance probably isn't exposed to the internet, instead, it's inside a VPC or similar. You'll have to either tunnel into the VPC via a Bastion, or run DuckDB on a EC2 machine inside the VPC. AWS usage isn't something we can provide support for however

**Reply by ramansh03** [reply]
@Mause But the thing is I was able to connect to the same redshift cluster using DuckDB python CLI but not able to connect from DuckDB CLI!

**Reply by Mause** [reply]
Given that I can't connect to the URL you've given (redshift-ingest-1.cfzy7kzssrme.ap-south-1.redshift.amazonaws.com), I assume that the instance has either been shut down or, like I mentioned above, it isn't exposed to the public internet, neither of which are things we can help with

**Reply by ramansh03** [reply]
@Mause I understand that you can't help me with this particular use case, But I am unaware of how can I connect with the same URL just now with the DuckDB Python CLI.
Also, the URL is up and running and I am able to connect to the same with DuckDB Python CLI and DBeaver.

**Reply by thengineer** [reply]
For what it's worth I also tried connecting to redshift using:
```
CALL postgres_attach('dbname=XXXX user=XXXX password=XXXX host=XXXX.us-west-2.redshift.amazonaws.com  port=5439');
```

I dont have firewall issues, but I do get this error: 
```
Error: IO Error: Unable to query Postgres: ERROR:  function pg_is_in_recovery() does not exist
HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
 ERROR:  function pg_is_in_recovery() does not exist
HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
```

I dont think the connection was established, because ` PRAGMA show_tables;` doesnt show my tables.
